### PR TITLE
Replace valkey in log and panic messages

### PR DESCRIPTION
**Pull request by Shivshankar-Reddy**
Part of #207

### CODE REVIEW

File: src/cluster_legacy.c [hunk 1]
```diff
@@ -546,7 +546,8 @@ int clusterLoadConfig(char *filename) {
             } else if (!strcasecmp(s, "noflags")) {
                 /* nothing to do */
             } else {
-                serverPanic("Unknown flag in redis cluster config file");
+                serverPanic("Unknown flag in %s cluster config file",
+                    server.extended_redis_compat ? "redis" : "valkey");
```
Comment (zuiderkwast) [inline src/cluster_legacy.c]: This config has limited scope and in general not used for logging. We want to delete it in the future so we don't want to add it everywhere. Please follow #207.

File: src/rdb.c [hunk 2]
```diff
@@ -3020,10 +3020,9 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
             if ((dbid = rdbLoadLen(rdb, NULL)) == RDB_LENERR) goto eoferr;
             if (dbid >= (unsigned)server.dbnum) {
                 serverLog(LL_WARNING,
-                          "FATAL: Data file was created with a Redis "
-                          "server configured to handle more than %d "
-                          "databases. Exiting\n",
-                          server.dbnum);
+                          "FATAL: Data file was created with a %s server configured to "
+                          "handle more than %d databases. Exiting\n",
+                          server.extended_redis_compat ? "Redis" : "Valkey", server.dbnum);
```
Comment (Shivshankar-Reddy) [inline src/rdb.c]: Hi @zuiderkwast , Do you think we have to update macro instead of config here as well?

File: src/cluster_legacy.c (hunk 1 above)
Comment (Shivshankar-Reddy) [reply src/cluster_legacy.c]: Added SERVER_NAME as it smention is #207.

File: src/cluster_legacy.c [hunk 3]
```diff
@@ -546,7 +546,7 @@ int clusterLoadConfig(char *filename) {
             } else if (!strcasecmp(s, "noflags")) {
                 /* nothing to do */
             } else {
-                serverPanic("Unknown flag in redis cluster config file");
+                serverPanic("Unknown flag in %s cluster config file", SERVER_NAME);
```
Comment (zuiderkwast) [inline src/cluster_legacy.c]: Better to use SERVER_TITLE here.

SERVER_TITLE = "Valkey" (uppercase "V")
SERVER_NAME = "valkey" (lowercase "v")

```suggestion
                serverPanic("Unknown flag in %s cluster config file", SERVER_TITLE);
```

File: src/rdb.c (hunk 2 above)
Comment (zuiderkwast) [reply src/rdb.c]: Yes, SERVER_TITLE. It was decided in #274 that this config should not affect logging.

```suggestion
                          SERVER_TITLE, server.dbnum);
```

**Review by zuiderkwast — APPROVED** [review]

**Comment by zuiderkwast** [comment]
Thanks!

**Pull request merged**